jina/helper.py

```diff
@@ -260,20 +260,33 @@ def random_name() -> str:
 
 
 def random_port() -> int:
+    from contextlib import closing
+    import socket
     if 'JINA_RANDOM_PORTS' not in os.environ:
         # feel like this gives higher chance of collision in unit test
-        from contextlib import closing
-        import socket
         import threading
         with threading.Lock():
             with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
                 s.bind(('', 0))
                 s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-                return s.getsockname()[1]
+                _port = s.getsockname()[1]
+                return _port
     else:
         import random
+
+        def is_port_in_use(p):
+            if p is None:
+                return True
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+                result = s.connect_ex(('', p))
+                s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                return result == 0
+
         min_port, max_port = 49152, 65535
-        return random.randrange(min_port, max_port)
+        _port = None
+        while is_port_in_use(_port):
+            _port = random.randrange(min_port, max_port)
```

Comment:
the coding style needs to be improved.

can't you set `_port = random.randrange(min_port, max_port)` first and then do the `while`? this remove `if p is None return True`